internal/controllers/machineset/machineset_controller.go

```diff
@@ -969,16 +993,54 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 
 func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
+
+	// Calculate how many in flight machines we should remediate.
+	// By default, we allow all machines to be remediated at the same time.
+	maxInFlight := len(filteredMachines)
+
+	// If the MachineSet is part of a MachineDeployment, only allow remediations if
```

Comment:
were you concern about any specific corner case where this check would help?

Reply:
There is generally no point in trying to remediate machines on old machinesets and having those to be scaled up again after the remediation happens. This solves that issue.